File: python/euler/bin/p0047.py
# project euler: problem 47

from itertools import count

from euler.lib.util import factorize


def compute(nfactors: int) -> str:
    cnt = 0
    for x in count(1):
        if len(factorize(x)) != nfactors:
            cnt = 0
        elif cnt == nfactors - 1:
            return str(x - (nfactors - 1))
        else:
            cnt += 1

    raise RuntimeError('unreachable!')


def solve() -> str:
    return compute(4)

File: typescript/euler/bin/p0047.ts
// project euler: problem 47

import { primeFactorization } from "../lib/factor.ts";

export const compute = (nfactors: number): string => {
  let cnt = 0;

  for (let x = 1;; x++) {
    if (primeFactorization(x).length !== nfactors) {
      cnt = 0;
    } else if (cnt === nfactors - 1) {
      return String(x - (nfactors - 1));
    } else {
      cnt += 1;
    }
  }
};

export const solve = (): string => compute(4);
